pinot-core/src/main/java/com/linkedin/pinot/core/operator/aggregation/DefaultAggregationExecutor.java

```diff
@@ -103,81 +92,16 @@ public void aggregate(ProjectionBlock projectionBlock) {
-        break;
 
-      case AggregationFunctionFactory.FASTHLL_AGGREGATION_FUNCTION:
-        aggregationFunction.aggregate(length, resultHolder,
-            (Object) blockValueSet.getSingleValues());
-        break;
-
-      case AggregationFunctionFactory.SUM_AGGREGATION_FUNCTION:
-      case AggregationFunctionFactory.AVG_AGGREGATION_FUNCTION:
-      case AggregationFunctionFactory.MAX_AGGREGATION_FUNCTION:
-      case AggregationFunctionFactory.MIN_AGGREGATION_FUNCTION:
-      case AggregationFunctionFactory.MINMAXRANGE_AGGREGATION_FUNCTION:
-      case AggregationFunctionFactory.PERCENTILE50_AGGREGATION_FUNCTION:
-      case AggregationFunctionFactory.PERCENTILE90_AGGREGATION_FUNCTION:
-      case AggregationFunctionFactory.PERCENTILE95_AGGREGATION_FUNCTION:
-      case AggregationFunctionFactory.PERCENTILE99_AGGREGATION_FUNCTION:
-      case AggregationFunctionFactory.PERCENTILEEST50_AGGREGATION_FUNCTION:
-      case AggregationFunctionFactory.PERCENTILEEST90_AGGREGATION_FUNCTION:
-      case AggregationFunctionFactory.PERCENTILEEST95_AGGREGATION_FUNCTION:
-      case AggregationFunctionFactory.PERCENTILEEST99_AGGREGATION_FUNCTION:
-        aggregationFunction.aggregate(length, resultHolder,
-            (Object) blockValueSet.getSingleValues());
-        break;
-
-      case AggregationFunctionFactory.SUM_MV_AGGREGATION_FUNCTION:
-      case AggregationFunctionFactory.AVG_MV_AGGREGATION_FUNCTION:
-      case AggregationFunctionFactory.MAX_MV_AGGREGATION_FUNCTION:
-      case AggregationFunctionFactory.MIN_MV_AGGREGATION_FUNCTION:
-      case AggregationFunctionFactory.MINMAXRANGE_MV_AGGREGATION_FUNCTION:
-      case AggregationFunctionFactory.PERCENTILE50_MV_AGGREGATION_FUNCTION:
-      case AggregationFunctionFactory.PERCENTILE90_MV_AGGREGATION_FUNCTION:
-      case AggregationFunctionFactory.PERCENTILE95_MV_AGGREGATION_FUNCTION:
-      case AggregationFunctionFactory.PERCENTILE99_MV_AGGREGATION_FUNCTION:
-      case AggregationFunctionFactory.PERCENTILEEST50_MV_AGGREGATION_FUNCTION:
-      case AggregationFunctionFactory.PERCENTILEEST90_MV_AGGREGATION_FUNCTION:
-      case AggregationFunctionFactory.PERCENTILEEST95_MV_AGGREGATION_FUNCTION:
-      case AggregationFunctionFactory.PERCENTILEEST99_MV_AGGREGATION_FUNCTION:
-        aggregationFunction.aggregate(length, resultHolder,
-            (Object) blockValueSet.getMultiValues());
-        break;
-
-      default:
-        throw new RuntimeException("Not supported function  " + aggrFuncName + aggrColumn);
+    if (!aggregationFunction.getName().equals(AggregationFunctionFactory.COUNT_AGGREGATION_FUNCTION)) {
+      Block dataBlock = projectionBlock.getDataBlock(aggregationColumns[0]);
+      ProjectionBlockValSet blockValueSet = (ProjectionBlockValSet) dataBlock.getBlockValueSet();
+      aggregationFunction.aggregate(length, resultHolder, blockValueSet);
+    } else {
+      aggregationFunction.aggregate(length, resultHolder);
```

Comment:
It seems we are first getting the result holder from aggregation function and then passing it back to the aggregation function. The 'get' misleads into thinking that result holder is actually sitting within aggr function, when you are probably using it in place of factory. You could still keep the factory and pass it the Aggr function object (instead of name) to get the appropriate result holder.